dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/ArgumentResolver.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.requests;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.Optional;
+
+import io.openliberty.mcp.internal.ToolMetadata.ArgumentMetadata;
+import io.openliberty.mcp.internal.schemas.TypeUtility;
+
+/**
+ * Resolves an argument value for a tool method parameter
+ */
+public class ArgumentResolver {
+
+    private static Map<Class<?>, Object> TYPE_DEFAULTS_MAP = Map.of(
+                                                                    boolean.class, false,
+                                                                    char.class, '\0',
+                                                                    byte.class, (byte) 0,
+                                                                    short.class, (short) 0,
+                                                                    int.class, 0,
+                                                                    long.class, 0L,
+                                                                    float.class, 0f,
+                                                                    double.class, 0d);
+
+    /**
+     * Tool Arguments are NOT required if:
+     *
+     * ToolArg.required is set to false
+     * ToolArg.defaultValue is set, and
+     * the argument return type is optional
+     */
+    public static boolean isRequiredArgument(ArgumentMetadata argMetadata) {
+
+        if (argMetadata.defaultValue() != null && !argMetadata.defaultValue().isEmpty()) {
+            return false;
+        }
+        if (isOptionalType(argMetadata.type())) {
+            return false;
+        }
+        return argMetadata.required();
+    }
+
+    public static Object resolveArgumentValue(ArgumentMetadata argMetadata,
+                                              String argName,
+                                              Map<String, Object> providedArgs) {
+
+        if (providedArgs.containsKey(argName)) {
+            return providedArgs.get(argName);
+        }
```

Comment:
```suggestion
        Object providedValue = providedArgs.get(argName);
        if (providedValue != null) {
            return providedValue;
        }
```
Don't call `containsKey` and `get` back to back like this.

Reply:
Done, thanks